junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClassFilter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.util.function.Predicate;
+
+import org.apiguardian.api.API;
+
+/**
+ * Class-related predicate holder used by reflection utilities.
+ *
+ * <h3>DISCLAIMER</h3>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
+ * @since 1.1
+ */
+@API(status = INTERNAL, since = "1.1")
+public class ClassFilter implements Predicate<Class<?>> {
+
+	public static ClassFilter of(Predicate<Class<?>> classPredicate) {
```

Comment:
Can we make the static factory methods more descriptive, e.g. `acceptAllNamesButFilterClasses()` and `filterNamesAndClasses()` or similar?

Reply:
Would a javadoc comment (stating the obvious) suffice? ;)